### DISCUSSION TITLE
Tor прокси

### DISCUSSION THREAD

**Discussion by Andrei030x** [anticensority/runet-censorship-bypass#187] [Q&A]
Расширение настроено на проксирование трафика через tor.
Оно прогоняет его через три узла или только через один выходной?
Просто я заметил что у расширения скорость гораздо выше, чем у tor браузера.

**Comment by ilyaigpetrov** [comment]
Расширение не поставляет встроенного Тора, а использует тот, что имеется в системе (запущенный как приложение с окном или как демон).
Соответственно, количество узлов настраивается тем, кто установил и запустил Тор в вашей ОС.

**Reply by Andrei030x** [reply]
Получается что это то же самое, как и если я с tor браузера буду сидеть. Только с похереной защитой.
Но откуда же такой лютый прирост скорости? Для тора это немыслимая скорость.
<img width="483" alt="1" src="https://user-images.githubusercontent.com/64521437/235945883-00878323-e664-4673-a007-e7839bf953dd.png">

**Reply by ilyaigpetrov** [reply]
Возможно, у вас запущено сразу два Тора: как приложение с окном и как служба/демон. Такое может быть более вероятным, если у вас Linux.

**Reply by Andrei030x** [reply]
Нее. Даж комп перезагрузил. И теперь с другим ip еще пинг 37 стал. Ваще прям чудеса твое расширение творит.
Ну и для верности: настроил другую прогу, что бы она так же, как твое расширение, через локал tor качала (socks 5). И уже только 15-30 мегабит удалось выжать. Эт конешн не оч точно, потому что ip всегда разные. Но в любом случае 90 мегабит с расширением эт слишком много.
Может ты попробуешь у себя это повторить? Только надо сайт в исключения закинуть.

**Reply by ilyaigpetrov** [reply]
*.speedtest.net добавлять? Какой PAC-скрипт? АнтиЦензорити?

**Reply by Andrei030x** [reply]
Да, все так.

**Reply by ilyaigpetrov** [reply]
https://www.speedtest.net/result/14690472969
![speedtest-tor](https://user-images.githubusercontent.com/1090811/236002774-97dfd72f-4d42-4261-aea9-7aff22fc074d.png)

**Reply by ilyaigpetrov** [reply]
Галочку "использовать прокси PAC-скрипта" не забывайте снять.

**Comment by Andrei030x** [comment]
ДА ТЫ РОФЛИШЬ. 5 МС ПИНГ https://www.speedtest.net/result/14690513105
![image](https://user-images.githubusercontent.com/64521437/236004947-3dbe1de2-f3b3-43c0-900b-aea1d1e09bc8.png)
Конешн галка снята. Причем можно [посмотреть](https://www.dan.me.uk/torlist/?exit), что эти ip - выходные узлы тора.

И как объяснить эту дичь?
В теории эти узлы тора могут быть шпионскими, у меня больше нет идей.

**Reply by Andrei030x** [reply]
А, я понял. У меня была эта очевидная идея с самого начала. Но раз ты про это не сказал, я подумал, что такого косяка быть не должно.
Если посмотреть по диспетчеру задач, то трафик идет через обычный браузер, а не через obfs4proxy. Видимо у сайта много адресов, и твое расширение их не все проксирует. И получается что отображается ip тора, а качается с моего ip напрямую. Хотя это и не косяк так то, ведь оно должно работать только для заблоченых.
Но меня сбил тот факт, что если смотреть видос например на [niconico](www.nicovideo.jp) (который в этой дыре заблочен), то с ним такой же прикол. Сайт проксируется, а сам видос и так напрямую проходит (через tor 1080p точно не посмотришь). Видимо у рабнадзора криво заблочено, ну в их духе.

**Reply by Andrei030x** [reply]
Ну и еще больше сбило, что у тебя то почему то все правильно работает. А у меня забагалось.
Напоминает ["ошибочку"](https://xakep.ru/2021/02/20/brave-onion-leak/) в говнобраузере brave, который в приватных окнах должен гнать трафик через сеть tor, но при этом сливал onion адреса DNS провайдеру в течении 3 лет...
Но у тебя хотя бы написано, что расширение не дает анонимности. Но вот [ссылка](https://rebrand.ly/ac-tor), по которой можно перейти из расширения, что бы это прочитать - у этих чертей уже ЗАБАНЕНА, исправляй. Нафига вообще эти редиректы, сразу бы [прямую](https://github.com/anticensority/runet-censorship-bypass/wiki/%D0%A0%D0%B0%D1%81%D1%88%D0%B8%D1%80%D0%B5%D0%BD%D0%B8%D0%B5-%D0%B8-TOR) на гитхаб указал.

**Reply by ilyaigpetrov** [reply]
> Нафига вообще эти редиректы[...]

Если мне вдруг захочется поменять заголовок страницы на wiki, то старая прямая ссылка ломается. Чтобы сменить прямую ссылку в расширении на новую, нужно пройти проверку в магазине, которая может занять недели 2. А у укороченной через rebrandly всегда можно сменить URL назначения, не прибегая к обновлению кода расширения.

В каком месте ты нашёл rebrand.ly/ac-tor? Во всплывающей подсказке расширения?

**Reply by Andrei030x** [reply]
Ну блин, а так ты дал им возможность ее легко сломать.
Эту тут
<img width="224" alt="2403" src="https://user-images.githubusercontent.com/64521437/236185652-4d230a2f-63ee-4dd5-b6ec-e13e22c9e722.png">
Но там у тебя много rebrand.ly ссылок. Я думаю тебе их легче все найти в коде.
Вот кста, почему расширение не проксирует rebrand.ly, когда он заблочен? Вообще с этим отдельную [тему](https://github.com/anticensority/runet-censorship-bypass/issues/188) создам.
